service/controller/resource/masters/create_master_instances_upgrading.go

```diff
@@ -25,26 +24,28 @@ func (r *Resource) masterInstancesUpgradingTransition(ctx context.Context, obj i
 		return "", microerror.Mask(err)
 	}
 
-	versionValue := map[string]string{}
+	r.Logger.LogCtx(ctx, "level", "debug", "message", "finding out if all tenant cluster master nodes are Ready")
 	{
-		for _, node := range cr.Status.Cluster.Nodes {
-			versionValue[node.Name] = node.Version
+		allMasterNodesAreReady, err := r.areNodesReadyForUpgrading(ctx)
+		if IsClientNotFound(err) {
+			r.Logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster client not available yet")
+			return currentState, nil
+		} else if err != nil {
+			return "", microerror.Mask(err)
 		}
-	}
 
-	var drainerConfigs []corev1alpha1.DrainerConfig
-	{
-		n := v1.NamespaceAll
-		o := metav1.ListOptions{
-			LabelSelector: fmt.Sprintf("%s=%s", label.Cluster, key.ClusterID(&cr)),
+		if !allMasterNodesAreReady {
+			r.Logger.LogCtx(ctx, "level", "debug", "message", "found out that all tenant cluster master nodes are not Ready")
```

Comment:
We only have one master node so this is technically true, but the reality is that we `found out that at least one master node is not ready`

Reply:
Good catch :+1: